test/powershell/Language/Scripting/ActionPreference.Tests.ps1

```diff
@@ -0,0 +1,124 @@
+﻿
+Describe "Tests for (error, warning, etc) action preference" -Tags "CI" {
+        BeforeAll {
+            $orgin = $GLOBAL:errorActionPreference
+        }
+
+        AfterAll {
+            if ($GLOBAL:errorActionPreference -ne $orgin)
+            {
+                $GLOBAL:errorActionPreference = $orgin
+            }
+        }
+        Context 'Setting ErrorActionPreference to stop prevents user from getting the error exception' {
+            $err = $null
+            try
+            {
+                get-childitem nosuchfile.nosuchextension -ea stop -ev err                
+            }
+            catch {}
+             
+            It '$err.Count' { $err.Count | Should Be 1 }
+            It '$err[0] should not be $null' { $err[0] | Should Not Be $null }
+            It '$err[0].GetType().Name' { $err[0].GetType().Name | Should Be ActionPreferenceStopException }
+            It '$err[0].ErrorRecord' { $err[0].ErrorRecord | Should not BeNullOrEmpty }
+            It '$err[0].ErrorRecord.Exception.GetType().Name' { $err[0].ErrorRecord.Exception.GetType().Name | Should Be ItemNotFoundException }
+        }
+        
+        It 'ActionPreference Ignore Works' {            
+            $errorCount = $error.Count
+            Get-Process -Name asdfasdfsadfsadf -ErrorAction Ignore            
+
+            $error.Count | Should Be $errorCount
+        }
+        
+        Context 'action preference of Ignore cannot be set as a preference variable' {           
+            try {
+                $GLOBAL:errorActionPreference = "Ignore"
+                Get-Process -Name asdfasdfasdf
+                Throw "Execution OK"
+             } catch {
+                     $_.CategoryInfo.Reason | Should Be NotSupportedException
+             }
+        }
+
+        It 'action preference of Suspend cannot be set as a preference variable' {            
+            try { 
+                    $GLOBAL:errorActionPreference = "Suspend"
+                    Get-Process -Name asdfasdfasdf
+                    Throw "Execution OK"
+                } catch { 
+                    $_.CategoryInfo.Reason | Should Be ArgumentTransformationMetadataException
+                }
+        }
+
+        It 'enum disambiguation works' {
```

Comment:
>t 'enum disambiguation works' [](start = 9, length = 29)

you've been changing the global state of errorActionPreference above - could that cause problems in this test?